common/java_deps.bzl

```
@@ -110,18 +110,24 @@ def _java_deps_impl(ctx):
     names = {}
     files = []
     filenames = []
+    outputPathOverrides = ctx.attr.java_deps_root_overrides
 
     mapping = ctx.attr.target[TransitiveJarToMavenCoordinatesMapping].mapping
 
-    for file in ctx.attr.target.data_runfiles.files.to_list():
+    for file in ctx.attr.target.data_runfiles.files.to_list() + ctx.attr.target.files.to_list():
         if file.extension == 'jar' and not file.path.startswith(LOCAL_JDK_PREFIX):
             if ctx.attr.maven_name and file.path not in mapping:
                 fail("{} does not have associated Maven coordinate".format(file.owner))
             filename = mapping.get(file.path, file.basename).replace('.', '-').replace(':', '-')
             if filename in filenames:
                 print("Excluded duplicate: {}".format(filename))
                 continue # do not pack JARs with same name
-            names[file.path] = ctx.attr.java_deps_root + filename + ".jar"
+            for jarPattern in outputPathOverrides:
+                if file.basename == jarPattern or (jarPattern.endswith("*") and file.basename.startswith(jarPattern.rstrip("*"))):
+                    names[file.path] = outputPathOverrides.get(jarPattern) + filename + ".jar"
```

@alexjpwalker `jarPattern` is guaranteed to exist:

```suggestion
                    names[file.path] = outputPathOverrides[jarPattern] + filename + ".jar"
```